azure-proto-network/SubnetBuilder.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Azure.ResourceManager.Core;
+
+namespace azure_proto_network
+{
+    /// <summary>
+    /// A class representing a builder object to help create a subnet.
+    /// </summary>
+    public class SubnetBuilder : ArmBuilder<Subnet, SubnetData>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SubnetBuilder"/> class.
+        /// </summary>
+        /// <param name="container"> The container to create the subnet in. </param>
+        /// <param name="subnet"> The data model representing the subnet to create. </param>
+        public SubnetBuilder(SubnetContainer container, SubnetData subnet)
```

Comment:
This is to be public? Is this to be replicated across other resources? Maybe I missed the discussion to add this. So now we have different ways for construction:
1.  new XXXData();  -> XXXContainer.Create()
2.  XXXContainer.Construct().Create()
3.  new XXXBuilder(), ->  builder.Create()

Reply:
good catch making it internal